pilot/pkg/networking/core/v1alpha3/route/route.go

```diff
@@ -472,3 +479,49 @@ func BuildDefaultHTTPRoute(clusterName string) *route.Route {
 		},
 	}
 }
+
+// translateFault translates networking.HTTPFaultInjection into Envoy's HTTPFault
+func translateFault(in *networking.HTTPFaultInjection) *xdshttpfault.HTTPFault {
+	if in == nil {
+		return nil
+	}
+
+	out := xdshttpfault.HTTPFault{}
+	if in.Delay != nil {
+		out.Delay = &xdsfault.FaultDelay{
+			Type:    xdsfault.FaultDelay_FIXED,
+			Percent: uint32(in.Delay.Percent),
+		}
+		switch d := in.Delay.HttpDelayType.(type) {
+		case *networking.HTTPFaultInjection_Delay_FixedDelay:
+			delayDuration := util.GogoDurationToDuration(d.FixedDelay)
+			out.Delay.FaultDelaySecifier = &xdsfault.FaultDelay_FixedDelay{
+				FixedDelay: &delayDuration,
+			}
+		default:
+			log.Warnf("Ignoring exponential fault in route")
+			out.Delay = nil
+		}
+	}
+
+	if in.Abort != nil {
+		out.Abort = &xdshttpfault.FaultAbort{
+			Percent: uint32(in.Abort.Percent),
+		}
+		switch a := in.Abort.ErrorType.(type) {
+		case *networking.HTTPFaultInjection_Abort_HttpStatus:
+			out.Abort.ErrorType = &xdshttpfault.FaultAbort_HttpStatus{
+				HttpStatus: uint32(a.HttpStatus),
+			}
+		default:
+			log.Warnf("Ignoring non http status type faults")
```

Comment:
same as above:

`Currently only "status" type HTTP faults are supported in routes`

or similar.